src/main/java/me/karakelley/http/controllers/StaticFilesController.java

```diff
@@ -1,30 +1,58 @@
 package me.karakelley.http.controllers;
 
-import me.karakelley.http.PublicDirectory;
+import me.karakelley.http.FileSystem.PublicDirectory;
 import me.karakelley.http.Request;
 import me.karakelley.http.Response;
 import me.karakelley.http.Status;
+import me.karakelley.http.ContentGeneration.ContentGenerator;
 
+import java.io.File;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.List;
 
 public class StaticFilesController implements Controller {
-  private final ArrayList availableActions = new ArrayList<>(Collections.singleton("GET"));
+
+  public final ArrayList availableActions = new ArrayList<>(Collections.singleton("GET"));
+  private final ContentGenerator contentGenerator;
+  private final PublicDirectory publicDirectory;
+
+  public StaticFilesController(PublicDirectory publicDirectory, ContentGenerator contentGenerator) {
+    this.contentGenerator = contentGenerator;
+    this.publicDirectory = publicDirectory;
+  }
 
   public Response respond(Request request) {
+    String requestedResource = request.getPath();
+
     Response response = new Response();
     if (request.validRequestLine() && availableActions.contains(request.getMethod())) {
-      return displayFilesResponse(response);
+      return serveDirectoryResponse(response, requestedResource);
     } else {
       return new InvalidRequestController().respond(request);
     }
   }
 
-  private Response displayFilesResponse(Response response) {
+  private Response serveDirectoryResponse(Response response, String requestedResource) {
     response.setStatus(Status.OK);
-    response.setBody(String.join("\n", PublicDirectory.getFiles()));
-    response.setHeaders("Content-Type", "text/plain");
+    response.setBody(String.join("", getFilesAndDirectoryLinks(requestedResource)));
+    response.setHeaders("Content-Type", "text/html");
     response.setHeaders("Content-Length", String.valueOf(response.getBody().getBytes().length));
     return response;
   }
+
+  private ArrayList<String> getFilesAndDirectoryLinks(String resource) {
+    List<File> files = publicDirectory.getDirectoriesAndFiles(resource);
+    ArrayList<String> resources = new ArrayList<>();
+    if (!files.isEmpty()) {
```

Comment:
Previously you had to do a null check here. The nice thing about changing `PublicDirectory#getDirectoriesAndFiles` to always return a `List<Files>` (and having that list potentially be empty) is that the client doesn't need to make this check at all any more—iterating over an empty list is simply a no-op.
```
List<File> files = new ArrayList<>();
for (File file : files ) {
  println("Hello")
}
```
Nothing gets printed.